翻译sbp和ddp (#420)
Oneflow-Inc/oneflow-documentation: en/docs/parallelism/05_ddp.md

In [Common Distributed Parallel Strategies](./01_introduction.md), we introduced the characteristics of data parallel.

Oneflow provides two ways to accomplish data parallel, and one of them is to use the original concept of Oneflow to run data parallel training by configurating consistent tensor. This is also the **recommanded way** to run data parallel training on Oneflow.

Besides, to facilitate the users who are transferring from Pytorch to Oneflow, Oneflow offers the interface consistent with Pytorch `torch.nn.parallel.DistributedDataParallel` , [oneflow.nn.parallel.DistributedDataParallel](https://oneflow.readthedocs.io/en/master/nn.html#oneflow.nn.parallel.DistributedDataParallel) so that users can also conveniently extend single machine training to data parallel training. 

> ## Run Data Parallel Training With Configuration

Reviewer: ```suggestion
## Run Data Parallel Training with SBP Configuration
```

> The following code runs data parallel training by configurating consistent tensor. 

Reviewer: ```suggestion
The following code runs data parallel training by configurating global tensor. 
```

原有的 “一致性视角/张量”，现在统一用 "全局视角/张量“了

??? code
    ```python
    import oneflow as flow
    import oneflow.nn as nn
    import flowvision
    import flowvision.transforms as transforms

    BATCH_SIZE=64
    EPOCH_NUM = 1

>     PLACEMENT = flow.placement("cuda",{0:[0,1]})

Reviewer: 接口没有更新。请统一检查更新

Author: 已解决

        transform=transforms.ToTensor(),
        download=True,
    )

    train_dataloader = flow.utils.data.DataLoader(
        training_data, BATCH_SIZE, shuffle=True
    )

    model = flowvision.models.mobilenet_v2().to(DEVICE)
    model.classifer = nn.Sequential(nn.Dropout(0.2), nn.Linear(model.last_channel, 10))
    model = model.to_consistent(placement=PLACEMENT, sbp=B)

    loss_fn = nn.CrossEntropyLoss().to(DEVICE)
    optimizer = flow.optim.SGD(model.parameters(), lr=1e-3)

    for t in range(EPOCH_NUM):
        print(f"Epoch {t+1}\n-------------------------------")
        size = len(train_dataloader.dataset)
        for batch, (x, y) in enumerate(train_dataloader):
            x = x.to_consistent(placement=PLACEMENT, sbp=S0)
            y = y.to_consistent(placement=PLACEMENT, sbp=S0)

            # Compute prediction error
            pred = model(x)
            loss = loss_fn(pred, y)

            # Backpropagation
            optimizer.zero_grad()
            loss.backward()
            optimizer.step()

            current = batch * BATCH_SIZE
            if batch % 5 == 0:
                print(f"loss: {loss:>7f}  [{current:>5d}/{size:>5d}]")
    ```

> We can see that this script is almost identical to the training script for the single-machine and single-card configuration, with a few exceptions on some lines related to consitent tensor set-ups. These codes are:

Reviewer: ```suggestion
We can see that this script is almost identical to the training script for the single-machine and single-device configuration, with a few exceptions on some lines related to global tensor set-ups. These codes are:
```

Author: 已解决

- Set placement to place the training one GPU 0 and 1: 

```python

>     PLACEMENT = flow.placement("cuda",{0:[0,1]})

Reviewer: 这个代码的接口还没改为最新，请统一检查、修改。
```suggestion
    PLACEMENT = flow.placement("cuda", [0,1])
```

Author: 已解决

```

- Broadcast the model on clusters:

```python
    model = model.to_consistent(placement=PLACEMENT, sbp=B)
```

- Split the data on cluster with `split(0)`: 

```python
    x = x.to_consistent(placement=PLACEMENT, sbp=S0)
    y = y.to_consistent(placement=PLACEMENT, sbp=S0)
```

> This allows us to follow the instructions in [Common Distributed Parallel Strategies](./01_introduction.md)

Reviewer: ```suggestion
This allows us to follow the instructions in [Common Distributed Parallel Strategy](./01_introduction.md)
```

应该这这个 PR 最上面的那个改动同步

## Run Data Parallel Training With DistributedDataParallel

> The following codes provides a quick start for training data parallel with `oneflow.nn.parallel.DistributedDataParallel` :

Reviewer: ```suggestion
The following codes provides a quick start for training data parallel with `oneflow.nn.parallel.DistributedDataParallel`:
```

       grad_fn=<accumulate_grad>)
```

> ### Codes

Reviewer: 对比下中文版，中文版貌似没这个三级标题了。

Author: 已解决
